school/src/main/java/ru/hogwarts/school/service/FacultyService.java

```diff
@@ -0,0 +1,51 @@
+package ru.hogwarts.school.service;
+
+import org.springframework.stereotype.Service;
+import ru.hogwarts.school.exception.FacultyNotFoundException;
+import ru.hogwarts.school.exception.StudentNotFoundException;
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.repository.FacultyRepository;
+
+import java.util.*;
+
+@Service
+public class FacultyService {
+    private final FacultyRepository facultyRepository;
+
+    public FacultyService(FacultyRepository facultyRepository) {
+        this.facultyRepository = facultyRepository;
+    }
+
+    public Faculty createFaculty(Faculty faculty) {
+        return facultyRepository.save(faculty);
+    }
+
+    public Faculty getFaculty(long id) {
+        if(facultyRepository.findById(id).isEmpty()){
+            throw new FacultyNotFoundException(id);
+        }
+        return facultyRepository.findById(id).get();
```

Comment:
Зачем 2 раза делать запрос в БД, если можно 1 раз сделать?
```suggestion
        if(facultyRepository.findById(id).isEmpty()){
            throw new FacultyNotFoundException(id);
        }
        return facultyRepository.findById(id).get();
```
```suggestion
        return facultyRepository.findById(id).orElseThrow(() -> new FacultyNotFoundException(id));
```